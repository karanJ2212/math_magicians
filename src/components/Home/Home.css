.home {
  display: flex;
  justify-content: center;
  align-content: center;
  flex-direction: column;
  padding: 70px 0;
  text-align: center;
}

h1 {
  text-align: center;
  color: rgb(26 33 50);
}

.desc {
  font-size: 1.3rem;
  font-weight: 400;
}
