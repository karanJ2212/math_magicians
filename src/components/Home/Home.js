import React from 'react';
import './Home.css';

export default function Home() {
  return (
    <div>
      <div className="home">
        <h1>Welcome</h1>
        <div className="desc">
          Math Magicians is a website for all fans of mathematics. Make simple
          calculations and view some of your favorite math quotes!
        </div>
      </div>
    </div>
  );
}
