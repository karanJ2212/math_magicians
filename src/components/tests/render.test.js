import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Home from '../Home/Home';
import Calculator from '../calculator/calculator';
import Navigation from '../Navbar/Navbar';
import Quote from '../quotes/quote';

it('Test if Home renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test if Calculator renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test if Quote renders correctly', () => {
  const tree = renderer.create(<Quote />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test if Home renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
