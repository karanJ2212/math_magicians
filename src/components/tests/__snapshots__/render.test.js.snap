// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Calculator renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="calText"
  >
    <h2>
      Lets do some math
    </h2>
  </div>
  <div
    className="calculator"
  >
    <div
      className="display-section"
    >
      <p
        className="display-value"
      >
        0  
         
      </p>
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="operator"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Test if Home renders correctly 1`] = `
<div>
  <div
    className="home"
  >
    <h1>
      Welcome
    </h1>
    <div
      className="desc"
    >
      Math Magicians is a website for all fans of mathematics. Make simple calculations and view some of your favorite math quotes!
    </div>
  </div>
</div>
`;

exports[`Test if Home renders correctly 2`] = `
<div
  className="navbar"
>
  <h1
    style={
      Object {
        "color": "aliceblue",
        "fontFamily": "Rubik Vinyl",
        "fontWeight": "bold",
      }
    }
  >
    Math Magician
  </h1>
  <ul>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        home
      </a>
    </li>
    <li>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        calculator
      </a>
    </li>
    <li>
      <a
        href="/quote"
        onClick={[Function]}
      >
        quote
      </a>
    </li>
  </ul>
</div>
`;

exports[`Test if Quote renders correctly 1`] = `
<div
  className=" pageContainer"
>
  <p
    style={
      Object {
        "fontFamily": "Tangerine",
      }
    }
  />
</div>
`;
